File: src/simula/infrastructure/llm/runtime/content.py
"""목적:
- LLM 응답 content를 공통 규칙으로 문자열화한다.
"""

from __future__ import annotations

from typing import Any


def content_to_text(content: Any) -> str:
    """LangChain content payload를 문자열로 정규화한다."""

    if isinstance(content, str):
        return content

    if isinstance(content, list):
        chunks: list[str] = []
        for item in content:
            if isinstance(item, str):
                chunks.append(item)
            elif isinstance(item, dict):
                text = item.get("text")
                if isinstance(text, str):
                    chunks.append(text)
        return "\n".join(chunks)

    return str(content)

File: src/backend/integrations/llm/content.ts
export function contentToText(content: unknown): string {
  if (typeof content === "string") {
    return content
  }
  if (Array.isArray(content)) {
    return content
      .map((part) => {
        if (typeof part === "string") {
          return part
        }
        if (typeof part === "object" && part && "text" in part) {
          return String(part.text)
        }
        return ""
      })
      .join("")
  }
  return ""
}

